elastic/kibana
# [SecuritySolution] Create Entity Engine status tab

## Summary

This ticket updates the Entity Store page to display tabs. The original content is inside the 'import entities tab'.
It also adds a new status tab. The tab allows users to visualize all installed engines and installed resources. That will help users monitor their engines and fix the errors.

* Add two tabs to the Entity Store page
  * The import entities tab has all the bulk upload content
  * The status tab has the new content created on this PR
* Move the "clear entity store data" button to the header according to design mockups.
* Delete unused stats route
* Rename `enablement` API docs to `enable`
* Add a new parameter to the status API (`withComponents`)
  * Should I make it snake cased?

### import entities tab
![Screenshot 2024-11-27 at 15 07 01](https://github.com/user-attachments/assets/c433e217-781e-4792-8695-2ee609efa654)


### status tab
![Screenshot 2024-11-27 at 15 07 20](https://github.com/user-attachments/assets/8970c023-22b3-4e83-a444-fa3ccf78ea42)


## How to test it
- Open security solution app with data
- Go to entity store page
- You shouldn't see the new tab because the engine is disabled
- Enable the engine and wait
- Click on the new tab that showed up
- It should list user and host engine components, and everything should be installed
- Delete or misconfigure some of the resources, the new status should be reflected on the tab.


## TODO:
- [x] Rebase main after https://github.com/elastic/kibana/pull/199762 is merged
  - [x] Remove temporary status hook
- [x] Fix the"clear entity data" button. It should re-fetch the status API.







### Checklist

Reviewers should verify this PR satisfies this list as well.
